basesandbox/src/ipc/domain_socket.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+
+use super::Ipc;
+use std::os::unix::net::UnixDatagram;
+use std::path::Path;
+
+pub struct DomainSocket {
+    address_src: String,
+    address_dst: String,
+    socket: UnixDatagram,
```

Comment:
Why did you use a datagram instead of a stream socket?
If you used a stream socket, you don't need to implement semaphore.

Reply:
@sgkim126 Because it's easier. Do you want me to rewrite this PR with stream?